packages/in_app_purchase/example/lib/main.dart

```diff
@@ -361,28 +365,32 @@ class _MyAppState extends State<MyApp> {
 
   void _listenToPurchaseUpdated(List<PurchaseDetails> purchaseDetailsList) {
     purchaseDetailsList.forEach((PurchaseDetails purchaseDetails) async {
+      await InAppPurchaseConnection.instance.consumePurchase(purchaseDetails);
```

Comment:
consumePurchase is only for Play Store.  This line should not be here when running on iOS or during error case as you won't be able to consume the purchaseDetails

Reply:
Good catch! This was added for a manual testing. It shouldn't be here.